multimodal/tests/unittests/test_matcher.py

```diff
@@ -121,3 +122,61 @@ def test_matcher(
             hyperparameters=hyperparameters,
             time_limit=30,
         )
+
+
+def test_text_semantic_search():
+    corpus = [
+        "A man is eating food.",
+        "A man is eating a piece of bread.",
+        "The girl is carrying a baby.",
+        "A man is riding a horse.",
+        "A woman is playing violin.",
+        "Two men pushed carts through the woods.",
+        "A man is riding a white horse on an enclosed ground.",
+        "A monkey is playing drums.",
+        "A cheetah is running behind its prey.",
+    ]
+    queries = [
+        "A man is eating pasta.",
+        "Someone in a gorilla costume is playing a set of drums.",
+        "A cheetah chases prey on across a field.",
+    ]
+
+    matcher = MultiModalMatcher(
+        pipeline="text_similarity",
+        hyperparameters={"model.hf_text.checkpoint_name": "sentence-transformers/all-MiniLM-L6-v2"},
+    )
+    hits = semantic_search(
+        matcher=matcher,
+        query_data=queries,
+        response_data=corpus,
+        top_k=5,
+    )
+    hits_gt = [
+        [
+            {"corpus_id": 0, "score": 0.7035943269729614},
+            {"corpus_id": 1, "score": 0.527172327041626},
+            {"corpus_id": 3, "score": 0.18880528211593628},
+            {"corpus_id": 6, "score": 0.10461556911468506},
+            {"corpus_id": 8, "score": 0.09811049699783325},
+        ],
+        [
+            {"corpus_id": 7, "score": 0.6432060599327087},
+            {"corpus_id": 4, "score": 0.2563084363937378},
+            {"corpus_id": 3, "score": 0.1389961689710617},
+            {"corpus_id": 6, "score": 0.11907944828271866},
+            {"corpus_id": 8, "score": 0.1079183965921402},
+        ],
+        [
+            {"corpus_id": 8, "score": 0.8252813816070557},
+            {"corpus_id": 0, "score": 0.13986822962760925},
+            {"corpus_id": 7, "score": 0.1292111724615097},
+            {"corpus_id": 6, "score": 0.10977005213499069},
+            {"corpus_id": 3, "score": 0.06506325304508209},
+        ],
+    ]
+
+    for per_query_hits, per_query_hit_gt in zip(hits, hits_gt):
```

Comment:
metrics are not tested here?

Reply:
Will add metric tests in later PRs. The tests may require install ir datasets.